deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/BaseRegistry.kt

```diff
@@ -12,31 +12,30 @@ import com.airbnb.deeplinkdispatch.base.MatchIndex
  */
 abstract class BaseRegistry(val registeredDeepLinks: List<DeepLinkEntry>,
                             /**
-                           * A binary match index, created by the annotation processor.
-                           * In a wrapper to make handling it easier
-                           */
-                          matchIndexArray: ByteArray) {
+                             * A binary match index, created by the annotation processor.
+                             * In a wrapper to make handling it easier
+                             */
+                            matchIndexArray: ByteArray,
+                            val pathSegmentKeysInRegistry: Set<String>) {
 
-    private val matchIndex: MatchIndex
-
-    init {
-        this.matchIndex = MatchIndex(matchIndexArray)
-    }
+    private val matchIndex: MatchIndex = MatchIndex(matchIndexArray)
 
     /**
      * Use a binary match index to match the given URL. Defaults can be find in [MatchIndex].
      *
      * @param deepLinkUri The input [DeepLinkUri] to match
      * @return A [DeepLinkEntry] if a match was found or null if not.
      */
-    fun idxMatch(deepLinkUri: DeepLinkUri?): DeepLinkEntry? {
+    @JvmOverloads
+    fun idxMatch(deepLinkUri: DeepLinkUri?, pathSegmentReplacements: Map<String, String> = mapOf()): DeepLinkEntry? {
```

Comment:
add kdoc param for `pathSegmentReplacements`?

Reply:
yes thanks!

Reply:
Bump